# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/base/TextLeafRange.cpp
# Commit: a62a2d25a53f
# Full Hash: a62a2d25a53f6f2ef0cfdcaeea61f7d6b9c8c284
# Author: James Teh <jteh@mozilla.com>
# Date: 2025-07-11 16:36:21
# Regressor Bug: 1950748
# File Overlap Count: 1
# Description:
#   Bug 1950748 part 2: Remove some dead code and add an assertion. r=morgan
#   
#   As of bug 1901457, the TextLeafPoint constructor can no longer receive TEXT_OFFSET_CARET.
#   However, a related check was never removed from the constructor.
#   Remove this check and assert that this offset is never provided.
# ==============================================================================

diff -r 17243f4cab40 -r a62a2d25a53f accessible/base/TextLeafRange.cpp
--- a/accessible/base/TextLeafRange.cpp	Fri Jul 11 06:10:56 2025 +0000
+++ b/accessible/base/TextLeafRange.cpp	Fri Jul 11 06:10:57 2025 +0000
@@ -637,6 +637,8 @@
 /*** TextLeafPoint ***/
 
 TextLeafPoint::TextLeafPoint(Accessible* aAcc, int32_t aOffset) {
+  MOZ_ASSERT(aOffset >= 0 ||
+             aOffset == nsIAccessibleText::TEXT_OFFSET_END_OF_TEXT);
   if (!aAcc) {
     // Construct an invalid point.
     mAcc = nullptr;
@@ -646,8 +648,7 @@
 
   // Even though an OuterDoc contains a document, we treat it as a leaf because
   // we don't want to move into another document.
-  if (aOffset != nsIAccessibleText::TEXT_OFFSET_CARET && !aAcc->IsOuterDoc() &&
-      aAcc->HasChildren()) {
+  if (!aAcc->IsOuterDoc() && aAcc->HasChildren()) {
     // Find a leaf. This might not necessarily be a TextLeafAccessible; it
     // could be an empty container.
     auto GetChild = [&aOffset](Accessible* acc) -> Accessible* {
